### PR TITLE
implement shell sort in python3

### PR DESCRIPTION
**Pull request by charindukavishan**
**Hi!**  
Thanks for considering for contributing to this ever-growing list of algorithm and data structure implementations.
Your contribution is valuable. Use the follow template for creating PRs. (Delete this part)
---

**Fixes #{ISSUE_NUMBER}**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [ ] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [ ] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [ ] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [ ] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [ ] This pull request shall only be merged once the build passes.

-   [ ] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    # Rearrange elements at each n/2, n/4, n/8, ... gaps
+    gap = n//2
+  
+    while gap > 0: 
+  
+        for i in range(gap,n): 
+            
+            temp = arr[i] 
+            j = i 
+            while  j >= gap and arr[j-gap] >temp:
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:14]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    
+    # Rearrange elements at each n/2, n/4, n/8, ... gaps
+    gap = n//2
+  
+    while gap > 0: 
+  
+        for i in range(gap,n): 
+            
+            temp = arr[i] 
+            j = i
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:13]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+def shellSort(arr): 
+    
+    n = len(arr)
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:3]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+def shellSort(arr): 
+
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:2]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
+def shellSort(arr):
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:1]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
+def shellSort(arr): 
+    
+    n = len(arr) 
+    
+    # Rearrange elements at each n/2, n/4, n/8, ... gaps
+    gap = n//2
+  
+    while gap > 0: 
+  
+        for i in range(gap,n):
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:10]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    gap = n//2
+  
+    while gap > 0: 
+  
+        for i in range(gap,n): 
+            
+            temp = arr[i] 
+            j = i 
+            while  j >= gap and arr[j-gap] >temp: 
+                arr[j] = arr[j-gap]
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:15]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
+def shellSort(arr): 
+    
+    n = len(arr) 
+    
+    # Rearrange elements at each n/2, n/4, n/8, ... gaps
+    gap = n//2
+  
+    while gap > 0:
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:8]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+  
+    while gap > 0: 
+  
+        for i in range(gap,n): 
+            
+            temp = arr[i] 
+            j = i 
+            while  j >= gap and arr[j-gap] >temp: 
+                arr[j] = arr[j-gap] 
+                j -= gap
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:16]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  
+        for i in range(gap,n): 
+            
+            temp = arr[i] 
+            j = i 
+            while  j >= gap and arr[j-gap] >temp: 
+                arr[j] = arr[j-gap] 
+                j -= gap 
+  
+            arr[j] = temp
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:18]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+    n = len(arr) 
+    
+    # Rearrange elements at each n/2, n/4, n/8, ... gaps
+    gap = n//2
+  
+    while gap > 0: 
+  
+        for i in range(gap,n): 
+            
+            temp = arr[i]
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:12]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+            j = i 
+            while  j >= gap and arr[j-gap] >temp: 
+                arr[j] = arr[j-gap] 
+                j -= gap 
+  
+            arr[j] = temp 
+        gap //= 2
+
+def main():
+    arr = [ 1,31,78,26,11,16,46]
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:22]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+            while  j >= gap and arr[j-gap] >temp: 
+                arr[j] = arr[j-gap] 
+                j -= gap 
+  
+            arr[j] = temp 
+        gap //= 2
+
+def main():
+    arr = [ 1,31,78,26,11,16,46] 
+    shellSort(arr)
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:23]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 14]
```diff
@@ -0,0 +1,21 @@
+def shellSort(arr):    
+    n = len(arr)     
+    # Rearrange elements at each n/2, n/4, n/8, ... gaps
+    gap = n//2  
+    while gap > 0:   
+        for i in range(gap,n):
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:6]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+            while  j >= gap and arr[j-gap] >temp: 
+                arr[j] = arr[j-gap] 
+                j -= gap   
+            arr[j] = temp 
+        gap //= 2
+
+def main():
+    arr = [ 1,31,78,26,11,16,46] 
+    shellSort(arr) 
+    print(arr)
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:18]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 16]
```diff
@@ -0,0 +1,21 @@
+def shellSort(arr):    
+    n = len(arr)     
+    # Rearrange elements at each n/2, n/4, n/8, ... gaps
+    gap = n//2  
+    while gap > 0:   
+        for i in range(gap,n):             
+            temp = arr[i] 
+            j = i
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:8]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 17]
```diff
@@ -0,0 +1,21 @@
+def shellSort(arr):    
+    n = len(arr)     
+    # Rearrange elements at each n/2, n/4, n/8, ... gaps
+    gap = n//2
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:4]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 18]
```diff
@@ -0,0 +1,21 @@
+def shellSort(arr):    
+    n = len(arr)     
+    # Rearrange elements at each n/2, n/4, n/8, ... gaps
+    gap = n//2  
+    while gap > 0:   
+        for i in range(gap,n):             
+            temp = arr[i] 
+            j = i 
+            while  j >= gap and arr[j-gap] >temp: 
+                arr[j] = arr[j-gap]
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:10]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 19]
```diff
@@ -0,0 +1,21 @@
+def shellSort(arr):    
+    n = len(arr)
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:2]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 20]
```diff
@@ -0,0 +1,21 @@
+def shellSort(arr):
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:1]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 21]
```diff
@@ -0,0 +1,21 @@
+def shellSort(arr):    
+    n = len(arr)     
+    # Rearrange elements at each n/2, n/4, n/8, ... gaps
+    gap = n//2  
+    while gap > 0:   
+        for i in range(gap,n):             
+            temp = arr[i] 
+            j = i 
+            while  j >= gap and arr[j-gap] >temp:
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:9]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 22]
```diff
@@ -0,0 +1,21 @@
+def shellSort(arr):    
+    n = len(arr)     
+    # Rearrange elements at each n/2, n/4, n/8, ... gaps
+    gap = n//2  
+    while gap > 0:
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:5]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 23]
```diff
@@ -0,0 +1,21 @@
...
+            j = i 
+            while  j >= gap and arr[j-gap] >temp: 
+                arr[j] = arr[j-gap] 
+                j -= gap   
+            arr[j] = temp 
+        gap //= 2
+
+def main():
+    arr = [ 1,31,78,26,11,16,46] 
+    shellSort(arr)
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:17]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 24]
```diff
@@ -0,0 +1,21 @@
...
+            temp = arr[i] 
+            j = i 
+            while  j >= gap and arr[j-gap] >temp: 
+                arr[j] = arr[j-gap] 
+                j -= gap   
+            arr[j] = temp 
+        gap //= 2
+
+def main():
+    arr = [ 1,31,78,26,11,16,46]
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:16]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 25]
```diff
@@ -0,0 +1,21 @@
...
+    n = len(arr)     
+    # Rearrange elements at each n/2, n/4, n/8, ... gaps
+    gap = n//2  
+    while gap > 0:   
+        for i in range(gap,n):             
+            temp = arr[i] 
+            j = i 
+            while  j >= gap and arr[j-gap] >temp: 
+                arr[j] = arr[j-gap] 
+                j -= gap
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:11]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 26]
```diff
@@ -0,0 +1,21 @@
...
+    # Rearrange elements at each n/2, n/4, n/8, ... gaps
+    gap = n//2  
+    while gap > 0:   
+        for i in range(gap,n):             
+            temp = arr[i] 
+            j = i 
+            while  j >= gap and arr[j-gap] >temp: 
+                arr[j] = arr[j-gap] 
+                j -= gap   
+            arr[j] = temp
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:12]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

File: algorithms/ar-shsrt/python3/shell_sort.py [hunk 27]
```diff
@@ -0,0 +1,21 @@
+def shellSort(arr):    
+    n = len(arr)     
+    # Rearrange elements at each n/2, n/4, n/8, ... gaps
+    gap = n//2  
+    while gap > 0:   
+        for i in range(gap,n):             
+            temp = arr[i]
```
Comment (RavinduSachintha) [inline algorithms/ar-shsrt/python3/shell_sort.py:7]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6323521)

**Review by Danushka96 — APPROVED** [review]

**Comment by Danushka96** [comment]
Hi @charindukavishan 
Thanks for the contribution, can you add your contribution to the algorithm chart in the main readme file. don't change anything else. And also push the changes to this PR.

**Comment by Danushka96** [comment]
@all-contributors please add @charindukavishan  for code and doc

**Comment by Danushka96** [comment]
@charindukavishan Thanks for the contribution, Keep it up (y)

**Pull request merged**